File: src/leetCode/Ex45.java
package leetCode;

public class Ex45 {
    public int jump(int[] nums) {
//        boolean[] reach = new boolean[nums.length];
        int res = 0;
        int temp = nums.length-1;
        while (temp>0){
            int t = temp;
            for (int i = temp-1;i>=0;i--){
                if (nums[i]+i>=temp){
                    t = i;
                }
            }
            temp=t;
            res++;
        }
        return res;
    }

    public static void main(String[] args) {
        System.out.println(new Ex45().jump(new int[]{2,3,1,1,4}));
    }
}

File: src/leetCode/ex45.kt
package leetCode

fun jump(nums: IntArray): Int {
    val res = IntArray(nums.size){0}
    (0 until res.size-1).forEach { base->
        (0 until nums[base]).forEach {
            res[base+it] = res[base]+1
        }
    }
    return res[nums.size-1]
}

fun main(args: Array<String>) {
    jump(intArrayOf(2,3,1,1,4)).apply { println(this) }
}